### ISSUE TITLE
epic(pixal3d): sparse-structure res 32/64 correctness + 4 GiB memory safety

### ISSUE THREAD

**Issue by raven38** [raven38/pixal3d.cpp#78]
## Goal

Make Pixal3D's sparse-structure resolution an explicit, validated runtime contract instead of the current implicit `ss_res=32` assumption, and support `ss_res=64` without reproducing the coordinate/OOM bug documented in visualbruno/ComfyUI-Trellis2#193.

Reference:
- https://github.com/visualbruno/ComfyUI-Trellis2/issues/193

Current `pixal3d.cpp` main does **not** hit that bug because the production Pixal3D path hardcodes `ss_coords(logits, 64, 32)` and uses sparse/device-resident projected conditioning. This epic is about making 64 a supported configuration safely rather than relying on that invariant forever.

## The correctness bug to avoid

For cascade sampling, decoder-upsampled coordinates span

```text
source_span = sparse_structure_resolution * 16
```

so:
- ss_res=32 -> source span 512
- ss_res=64 -> source span 1024

Any quantizer that always divides by 512 is correct only for `ss_res=32`. At 64 it doubles coordinates, can move the 1024-shape-flow grid from ~64 to ~128, and can trigger an 8x dense R^3 allocation if downstream conditioning derives its grid from coord maxima.

The invariant must be:

```text
normalized_coord = (upsampled_coord + 0.5) / (ss_res * 16)
target_coord     = normalized_coord * target_grid_scale
```

with Pixal3D's own verified round/grid-1 convention preserved where applicable.

### A. WebGPU / GPU single-buffer problem

The old dense paths are unsafe:
- dense projected conditioning at R=64: `64^3 * 2048 * fp32 = 2 GiB`
- positive+negative copies can approach/exceed 4 GiB
- NAF@1024 full output: `[1024, 1024^2]` fp32 = 4 GiB

The current production path already has the right direction:
- `pixal3d_cond_slat_gpu(..., &active_coords)` returns only active sparse rows
- NAF@1024 is built/consumed in chunks
- `trellis_cli.cpp` explicitly avoids materializing the 4 GiB dense NAF map

This epic must preserve those properties at `ss_res=64`; no fallback may silently recreate dense R^3 or full NAF buffers.

### B. wasm32 host-address-space problem

This is **not solved** by sparse GPU conditioning.

The browser build still has a structural ~4 GiB wasm32 linear-memory ceiling. Dense res-1024 meshes can still fail in decode/remesh/TriBvh/QEM/postprocess with `std::bad_alloc`. Existing related work:
- #38 desktop VRAM tracker
- #42 WebGPU C2S tiling / smaller maxBufferSize adapters
- `docs/PIXAL3D_E2E_STATUS.md`
- `docs/PIXAL3D_WEBGPU_MEMORY.md`

This epic therefore has separate acceptance gates for GPU buffers and wasm32 host memory.

## Required behavior

- default remains `ss_res=32`; bit/numerical behavior must not regress
- explicit `ss_res=64` support for Pixal3D cascade
- no hardcoded `/512` assumption where the source span depends on `ss_res`
- LR projected conditioning grid follows the actual sparse coord domain
- HR grid is chosen from the model/target resolution, **not** inferred from an accidentally inflated coord maximum
- sparse conditioning stays sparse at both 32 and 64
- no single NAF/projection buffer crosses the adapter's reported `maxBufferSize` / `maxStorageBufferBindingSize`
- browser host-memory failures are either eliminated or fail early with a documented bounded fallback; never hang/crash silently

## Dependency DAG

```text
#79 reference oracle / fixtures
  |
  v
#80 parameterize ss_res + audit hardcoded 32/512
  |
  v
#81 ss_res-aware cascade quantization
  |
  +-------------> #82 sparse conditioning / WebGPU per-buffer gate
  |
  +-------------> #83 wasm32 host-heap bounded tail
                    |
                    v
#79 + #80 + #81 + #82 + #83
                    |
                    v
                  #84 final E2E release gate
```

#82 and #83 intentionally separate device-buffer limits from wasm32 host memory.

## Sub-issues / execution order

| issue | task | dependency |
|---|---|---|
| [ ] #79 | pin ss_res=32/64 Python/reference oracle + fixtures | root |
| [ ] #80 | parameterize sparse-structure resolution; remove hardcoded ss_res=32 assumptions | #79 |
| [ ] #81 | make cascade coordinate normalization ss_res-aware | #79, #80 |
| [ ] #82 | keep ss_res=64 conditioning sparse; enforce WebGPU/GPU per-buffer limits | #80, #81 |
| [ ] #83 | bound decode/postprocess within wasm32 4 GiB host heap | can investigate in parallel; required for browser release gate |
| [ ] #84 | final 32/64 E2E quality + memory + backend matrix | #79–#83 |

Related existing memory work:
- #38 desktop 8 GB VRAM tracking
- #42 C2S tiling for smaller WebGPU maxBufferSize adapters

## Parent acceptance

- [ ] ss_res=32 regression oracle passes
- [ ] ss_res=64 reference/parity fixture passes
- [ ] cascade coordinate normalization is resolution-aware
- [ ] all hardcoded 32/512 assumptions in the Pixal3D path are audited
- [ ] R=64 conditioning never materializes dense R^3 projected conditioning
- [ ] NAF@1024 never materializes one 4 GiB output tensor in the production path
- [ ] WebGPU buffer-limit gate tested on at least the measured ~4 GiB and ~2 GiB binding-limit classes
- [ ] wasm32 host-memory postprocess has a bounded/validated policy
- [ ] E2E memory/perf/quality matrix recorded for ss_res=32 vs 64
- [ ] docs clearly distinguish WebGPU buffer limits, aggregate VRAM, and wasm32 host address space
